### DISCUSSION TITLE
[Done] Update MongoDB Node.js driver to v4.12.1 to improve performance

### DISCUSSION THREAD

**Discussion by dengribar** [meteor/meteor#12332] [Feature Requests]
👋 There was a noticeable performance regression in MongoDB Node.js driver **v4** in comparison with **v3** ([NODE-4283](https://jira.mongodb.org/browse/NODE-4283)), which seems to be resolved in **v4.12.0**, and announced as "**Performance Improvements with Buffering**" ([ref](https://github.com/mongodb/node-mongodb-native/releases/tag/v4.12.0#:~:text=Performance%20Improvements%20with%20Buffering)).

I was able to reproduce the improvement with the a [load test](https://github.com/dengribar/loadtest-mongodb-node-driver):

<img width="504" alt="image" src="https://user-images.githubusercontent.com/111731892/204788451-a685d188-e521-4b14-9c46-2b69790e48d9.png">

<img width="501" alt="image" src="https://user-images.githubusercontent.com/111731892/204788488-42c15ff0-3ee1-4e88-8a4d-fa40ca6b11f1.png">

So, considering there were no breaking changes between the current driver version (**4.11.0**) used internally by Meteor, and the latest version **4.12.1**, it would be awesome if Meteor could update the driver to the latest version as soon as possible. Thanks!

**Comment by afrokick** [comment]
@Grubba27 2.8.3?)

**Reply by Grubba27** [reply]
It is in 2.9-rc.1. Probably will launch an official 2.9 soon. Since it is RC, it can be used no problem, I think the RC and the official will look the same
